syntax = "proto3";

package adaptive_engine;
option go_package = "github.com/bl4ck0w1/deepFuzz/proto/gen/adaptive_engine";

service AdaptiveEngine {
  rpc Prioritize(PrioritizeRequest) returns (PrioritizeResponse) {};
  rpc Evaluate(EvaluateRequest) returns (EvaluateResponse) {};
  rpc Feedback(FeedbackRequest) returns (FeedbackResponse) {};
}


message Request {
  string url = 1;
  int32 depth = 2;
  int32 static_score = 3;
  float dynamic_score = 4;
  int32 retries = 5;
  string source = 6;
  int64 discovered = 7;  
}

message PrioritizeRequest {
  repeated Request requests = 1; 
  string cluster_id = 2;  
}

message PrioritizeResponse {
  string next_target = 1;
  float confidence = 2;
}

message EvaluateRequest {
  string url = 1;
  int32 status_code = 2;
  bytes response_snippet = 3;  
  string content_hash = 4;     
  string response_hash = 5;  
}

message EvaluateResponse {
  float score = 1;
}

message FeedbackRequest {
  string path = 1;
  bool is_critical = 2;
  bool is_false_positive = 3;
}

message FeedbackResponse {
  bool ack = 1;
}
